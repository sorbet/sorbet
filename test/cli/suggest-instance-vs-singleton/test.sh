#!/bin/bash

tmp="$(mktemp -d)"
infile="test/cli/suggest-instance-vs-singleton/suggest-instance-vs-singleton.rb"
cp "$infile" "$tmp"

cwd="$(pwd)"
cd "$tmp" || exit 1

"$cwd/main/sorbet" --silence-dev-message -a --suggest-unsafe suggest-instance-vs-singleton.rb 2>&1

echo
echo --------------------------------------------------------------------------
echo

# Also cat the file, to make sure that the autocorrect applied
cat suggest-instance-vs-singleton.rb

rm suggest-instance-vs-singleton.rb

