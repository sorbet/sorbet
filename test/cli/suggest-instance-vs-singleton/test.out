suggest-instance-vs-singleton.rb:8: Method `my_singleton_class_method` does not exist on `A` https://srb.help/7003
     8 |    my_singleton_class_method
            ^^^^^^^^^^^^^^^^^^^^^^^^^
  There is a singleton class method with the same name:
    suggest-instance-vs-singleton.rb:5: Defined here
     5 |  def self.my_singleton_class_method; end
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    Either:
    - use `.class` to call it,
    - remove `self.` from its definition to make it an instance method, or
    - define the current method as a singleton class method using `def self.`
  Autocorrect: Done
    suggest-instance-vs-singleton.rb:8: Inserted `self.class.`
     8 |    my_singleton_class_method
            ^

suggest-instance-vs-singleton.rb:11: Method `my_instance_method` does not exist on `T.class_of(A)` https://srb.help/7003
    11 |    my_instance_method
            ^^^^^^^^^^^^^^^^^^
  Autocorrect: Done
    suggest-instance-vs-singleton.rb:11: Replaced with `instance_method`
    11 |    my_instance_method
            ^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/master/rbi/core/module.rbi#L1048: Defined here
    1048 |  def instance_method(arg0); end
            ^^^^^^^^^^^^^^^^^^^^^^^^^
Errors: 2

--------------------------------------------------------------------------

# typed: true

class A
  def my_instance_method; end
  def self.my_singleton_class_method; end

  def example1
    self.class.my_singleton_class_method
  end
  def self.example2
    instance_method
  end
end
